UI/src/view/point_asset/trafficSignForm.js

```diff
@@ -413,19 +431,57 @@
         '      </select></div>';
     };
 
-    var singleChoiceTrafficSignTypeHandler = function (property, collection) {
+
+    function isPedestrianOrCyclingRoadLink(asset) {
+      if (asset === undefined){
+        return false;
+      }
+
+      var roadLink = me.roadCollection.getRoadLinkByLinkId(asset.linkId);
+      var roadLinkData;
+
+      if (roadLink){
+        roadLinkData = roadLink.getData();
+        if (roadLinkData !== undefined && roadLinkData.linkType === 8) {
+          return true;
+        }
+      }
+      return false;
+    }
+
+    var singleChoiceTrafficSignTypeHandler = function (property, collection, asset) {
       var propertyValue = (property.values.length === 0) ? '' : _.head(property.values).propertyValue;
       var signTypes = _.map(_.filter(me.enumeratedPropertyValues, { 'publicId': property.publicId}), function(val) {return val.values; });
+      var auxProperty = property;
       var groups =  collection.getGroup(signTypes);
       var groupKeys = Object.keys(groups);
       var mainTypeDefaultValue = _.indexOf(_.map(groups, function (group) {return _.some(group, function(val) {return val.propertyValue == propertyValue;});}), true);
+
+      if (isPedestrianOrCyclingRoadLink(asset)) {
+        var mandatorySignsDefaultValue = "70";
+        /* get the correct index for group to be used in subSingleChoice */
+        mainTypeDefaultValue = _.indexOf(_.map(groups, function (group) {return _.some(group, function(val) {return val.propertyValue == mandatorySignsDefaultValue;});}), true);
+
+        /* Only after get the correct index of the group (mainTypeDefaultValue) I can reset the values for the */
+        /* correct one to be used in the 'main singleChoice field' */
+        signTypes = _.map(signTypes,function(sign) { return _.filter(sign, function(x) { if ( x.propertyValue == '70') return x; });  });
+        groups =  collection.getGroup(signTypes);
+        groupKeys = Object.keys(groups);
```

Comment:
Nothing "bad" here just to say that we can abuse the lodash library a bit more (in my opinion) so Object.keys(groups) can be changed to _.keys(groups)

Reply:
Done